File: 2026/07-July/12-Sunday/solution.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] sorted = arr.clone();
        Arrays.sort(sorted);

        HashMap<Integer, Integer> rank = new HashMap<>();

        for (int x : sorted) {
            rank.putIfAbsent(x, rank.size() + 1);
        }

        int[] ans = new int[arr.length];
        for (int i = 0; i < arr.length; i++) {
            ans[i] = rank.get(arr[i]);
        }

        return ans;
    }
}

File: 2026/07-July/12-Sunday/solution.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> vals = arr;
        sort(vals.begin(), vals.end());
        vals.erase(unique(vals.begin(), vals.end()), vals.end());

        unordered_map<int, int> rank;
        for (int i = 0; i < vals.size(); i++)
            rank[vals[i]] = i + 1;

        vector<int> ans;
        ans.reserve(arr.size());

        for (int x : arr)
            ans.push_back(rank[x]);

        return ans;
    }
};
